src/utils/ddb/ddb_vos.c

```diff
@@ -2040,3 +2048,330 @@ dv_dev_replace(const char *db_path, uuid_t old_devid, uuid_t new_devid)
+		rc = smd_rdb_get_blob_sz(pool_info->spi_id, &rdb_blob_sz);
+		if (rc && (rc != -DER_NONEXIST)) {
+			D_ERROR(DF_UUID ": failed to extract the size of rdb file: " DF_RC "\n",
+				DP_UUID(pool_info->spi_id), DP_RC(rc));
+			goto out;
+		}
+
+		D_ASSERT(pool_info->spi_scm_sz > 0);
+		rc = vos_pool_recreate_tgt(pool_info->spi_id, pool_info->spi_scm_sz,
+					   pool_info->spi_tgt_cnt[SMD_DEV_TYPE_META], rdb_blob_sz,
+					   scm_mount, NULL);
+		if (rc)
+			goto out;
+	}
+out:
+	d_list_for_each_entry_safe(pool_info, tmp, &pool_list, spi_link) {
+		d_list_del(&pool_info->spi_link);
+		/* Frees spi_tgts, spi_blobs, and pool_info */
+		smd_pool_free_info(pool_info);
+	}
+	return rc;
+}
+
+static int
+ddb_tmpfs_setup(const char *scm_mount)
+{
+	int    rc = 0;
+	mode_t stored_mode;
+	char   newborns_path[DDB_SETUP_BUF_MAX] = {0};
+	char   zombies_path[DDB_SETUP_BUF_MAX]  = {0};
+	/** create the path string */
+	snprintf(newborns_path, sizeof(newborns_path), "%s/NEWBORNS", scm_mount);
+	snprintf(zombies_path, sizeof(zombies_path), "%s/ZOMBIES", scm_mount);
+
+	stored_mode = umask(0);
+	/** create NEWBORNS directory if it does not exist already */
+	rc = mkdir(newborns_path, S_IRWXU);
+	if (rc < 0 && errno != EEXIST) {
+		D_ERROR("failed to create NEWBORNS dir: %d\n", errno);
+		umask(stored_mode);
+		return daos_errno2der(errno);
+	}
+
+	/** create ZOMBIES directory if it does not exist already */
+	rc = mkdir(zombies_path, S_IRWXU);
+	if (rc < 0 && errno != EEXIST) {
+		D_ERROR("failed to create ZOMBIES dir: %d\n", errno);
+		umask(stored_mode);
+		return daos_errno2der(errno);
+	}
+	umask(stored_mode);
+	/** clear remain dir/file */
+	ddb_clear_dir(newborns_path);
+	ddb_clear_dir(zombies_path);
+
+	recreate_pooltgts(scm_mount);
+	if (rc != 0)
+		D_ERROR("Failed to recreate vos/rdb files. " DF_RC "\n", DP_RC(rc));
+
+	return rc;
+}
+
+int
+dv_run_setup(const char *db_path, const char *scm_mount, unsigned int scm_mount_size)
+{
+	int          rc;
+	bool         md_on_ssd;
+	bool         need_mount                 = true;
+	unsigned int sz                         = scm_mount_size;
+	char         options[DDB_SETUP_BUF_MAX] = {0};
+
+	rc = vos_self_init(db_path, true, 0);
+	if (rc) {
+		D_ERROR("Initialize standalone VOS failed. " DF_RC "", DP_RC(rc));
+		return rc;
+	}
+
+	md_on_ssd = bio_nvme_configured(SMD_DEV_TYPE_META);
+	if (!md_on_ssd) {
+		D_ERROR("Not md on ssd mode, Ignore to setup.");
+		goto out;
+	}
+
+	/* Fetch scm_mount_size */
+	if (sz == 0) {
+		rc = auto_calculate_scm_mount_size(&sz);
+		if (rc) {
+			D_ERROR("Failed to calculate scm size. " DF_RC "", DP_RC(rc));
+			goto out;
+		}
+		D_INFO("Not specify scm size. Automatically calculated size %uGiB", sz);
+	}
+
+	/* Make sure mountpoint is clear */
+	rc = tmpfs_setup_precheck(scm_mount, sz, &need_mount);
+	if (rc) {
+		D_ERROR("Failed to check scm mountpoint %s. " DF_RC "", scm_mount, DP_RC(rc));
+		goto out;
+	}
```

Comment:
I am not sure this bit is necessary as well. If the DDB operator provided a path it is their responsibility to make sure it makes sense.

I am a little bit worried this implementation is really overcomplicated.

Do we really need this?